python/llm/src/bigdl/llm/transformers/models/yuan.py

```diff
@@ -20,13 +20,32 @@
 # https://huggingface.co/IEITYuan/Yuan2-2B-hf/blob/7ab7b3c18eb8e5232ce2a3f720d4e6f4b53a2806/README.md#%E5%A3%B0%E6%98%8E%E4%B8%8E%E5%8D%8F%E8%AE%AEterms-and-conditions
 #
 
-import torch
 import copy
 import math
+from einops import rearrange
 from typing import Optional, Tuple
 
-from bigdl.llm.transformers.models.utils import apply_rotary_pos_emb
+import torch
+import torch.nn as nn
 from bigdl.llm.utils.common import invalidInputError
+from bigdl.llm.transformers.models.utils import apply_rotary_pos_emb
+from bigdl.llm.transformers.models.utils import init_kv_cache, extend_kv_cache, append_kv_cache
+from bigdl.llm.transformers.models.utils import is_enough_kv_cache_room_4_31
+from bigdl.llm.transformers.low_bit_linear import SYM_INT4, FP8E5
+
+KV_CACHE_ALLOC_BLOCK_LENGTH = 256
+
+
+def use_decoding_fast_path(q_type, use_fuse_rope, enough_kv_room, bs):
+    return q_type in [SYM_INT4, FP8E5] and \
+        use_fuse_rope and enough_kv_room and bs == 1
+
+
+def should_use_fuse_rope(self, hidden_states, position_ids):
```

Comment:
maybe we can put `use_decoding_fast_path` and `should_use_fuse_rope` into `utils` later,.